### ISSUE TITLE
The plugin should allow for downloading the CLI binary from a custom location

### ISSUE THREAD

**Issue by bmuschko** [snyk/gradle-plugin#4]
Some organizations do not want to reach out to the internet for security and reliability reasons in order to download the executable. Instead they want to host the executable in an inhouse repository of some sort. The plugin should allow for providing a custom location to download the executable from there instead of the hard-coded URLs.

https://github.com/snyk/gradle-plugin/blob/master/src/main/java/io/snyk/gradle/plugin/CliDownloader.java#L24-L25

**Comment by bmvermeer** [comment]
Not entirely true,
If the cli is installed already and available system-wide it will not download the cli.

If there is no cli available on the system it will download the latest cli.
Depending on the config the cli will be upgraded or not. If there is already a cli binary available for this project, this binary will be reused unless there is a newer version (depending on the config.

**Comment by bmvermeer** [comment]
Good point, this first edition was build for open-source uses not so much from a company pov.
I will mark it as an enhancement.
As it is a community project feel free to commit a PR yourself.
